### PR TITLE
dtls.c, netq.c: consider 32bit time overflow.

### PR DESCRIPTION
**Pull request by boaks**
Uses proposal in issue #125 to fix a 32bit overflow.

Fixes issue: #125

Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

File: dtls_time.h [hunk 1]
```diff
@@ -73,6 +75,11 @@ typedef clock_time_t dtls_tick_t;
...
...gh9E0YhyGCKh6GYdhyy4fhhHEaRyzLORyCUfBgV0bpuzptBWasRZwocRAJasXx3G/oJ9LCRBYmZelkn9oOw6jsQ46oJO7wznODALvpHnCppVa5Z2%2BkBoZh4mSe5m%2BZZ1kQE5341si4k9Z2XkiSJk0rv56WbZqSEoWh8XhbWUXLARuyxWRFEDZtQV7eh0VESRF2Ph6AGvuGV3QcQb6zCabG/DRHBTLQnAIrwngcFopCoJwv4KDMcx6ncPCkAQmiA1M5IgKsqwnNjeP4wTMTAxwkhg2jUOcLwCggN4qMQ4DpBwLASBoCwsR0BE5CUKz7P0JEyDAFwXAJTQtAEBE1MQKE5OhAEVTouTrNsIIYIMLQCv06QWCskY4ia/gX1lKO1Oa5gqilK44vk%2B8mDE5DtB4KExDy84WCcMjBDISw7sM%2BOboKAAangmB9mCWLg8j/CCCIYjsFIMiCIoKjqJruhcPohgmGY%2BiO9TkBTFhNgCCb%2B5gqsVO26UReeBADgDPUPidaMXSROn8SJNX9d6O32QMM3BTdOnTTV60/QuHUejD%2BUfTtAEnQD63wxj2kDeHCMc9jIPUxw7M8x6J7mALDwQMg2TmvQxwqj4lEe5RJI2zAMg2ZCyckUQI45G4IQJC7Gs6fbM4NmHMixIwmLwOmWgJgYyxhoV%2BXA%2BTY0kKsfEGg%2BTLD5AATlWNIYmpNSDg0hhfKmNMUZoymEzKAMBEAgCIC4LmEAqjABNrwRhChlCGFtkIBALUfakB5sAvQ2Jv6LyjsIUQ4h46iKTmocmLABBMDQF/IgxAAASTAFB0kPgwa4ltUBh0YNiXhcjmCKPwMo/REdeDGIUagCo%2BBLG9HsZTRxAR2GBFoFwlqDjaG0AAJLoBALEJg3D6CkD7M7WIPsT4cFBvg8mF8gkhMRn/aShAEDbCvjfO%2BD8n4Ji4K/cBZDoHY1xgTMp2N9CcDwd7JEcTz7OOprTMhjMqEQCQDMAgsRLb0P4XzQRZiSB6FETHCR0gpFKBkZrcJTBIncAZsTWJBDeAXzBJbLpxJvQZOvrfe%2Bj9n75LfoA3mERf6rETIU%2BmUDSAIEwEwLAkQIDROqSAWpSyKYcGIU0y5xSognEkJILBqx4r4gRFg/ECDKkcHLnUwhziIHo0hcsM%2BsKPmkO%2BaQRqiQ7CSCAA%3D
```
Comment (boaks) [inline dtls_time.h:78]: Maybe, I better remove the "huge" link ...

Comment (kfessel) [reply dtls_time.h:78]: godbolt also has short links 
https://godbolt.org/z/YchexKaeT

Comment (boaks) [reply dtls_time.h]: Much better.

File: dtls_time.h [hunk 2]
```diff
@@ -73,6 +75,11 @@ typedef clock_time_t dtls_tick_t;
 void dtls_clock_init(void);
 void dtls_ticks(dtls_tick_t *t);
 
+/* see https://godbolt.org/z/YchexKaeT */
+#define DTLS_OFFSET_TIME (((clock_time_t)~0) >> 1)
+/** Checks if A is before (or equal) B. Considers 32 bit time overflow */
+#define DTLS_IS_BEFORE_TIME(A, B) ((DTLS_OFFSET_TIME + (B)-(A)) >= DTLS_OFFSET_TIME)
```
Comment (kfessel) [inline dtls_time.h:81]:
```suggestion
#define DTLS_IS_BEFORE_TIME(A, B) ((clock_time_t)(DTLS_OFFSET_TIME + (B)-(A)) >= DTLS_OFFSET_TIME)
```
that cast is needed (check with godbolt)

Comment (boaks) [reply dtls_time.h]: Thanks for pointing to that.

**Comment by obgm** [comment]
Nice, thanks @boaks and @kfessel

**Pull request merged**